File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/inheritance/food_product.py
from product import Product

class FoodProduct(Product):
    def __init__(self, productId, name, price, manufacturer, expirationDate, isOrganic):
        super().__init__(productId, name, price, manufacturer)
        self._expirationDate = expirationDate
        self._isOrganic = isOrganic

    # Getter and Setter for the additional field
    def getExpirationDate(self):
        return self._expirationDate

    def setExpirationDate(self, expirationDate):
        self._expirationDate = expirationDate

    def isOrganic(self):
        return self._isOrganic

    def setOrganic(self, organic):
        self._isOrganic = organic

    # Override the displayInfo method to include additional information
    def displayInfo(self):
        super().displayInfo()
        print(f"Expiration Date: {self._expirationDate}, Organic: {self._isOrganic}")

File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/Inheritance/FoodProduct.java
import java.time.LocalDateTime;

class FoodProduct extends Product {
    // Additional field
    private LocalDateTime expirationDate;
    private boolean isOrganic;

    public FoodProduct(int productId, String name, double price, String manufacturer, LocalDateTime expirationDate, boolean isOrganic) {
        super(productId, name, price, manufacturer); // Call the superclass constructor
        this.expirationDate = expirationDate;
        this.isOrganic = isOrganic;
    }

    // Getter and Setter for the additional field
    public LocalDateTime getExpirationDate() {
        return expirationDate;
    }

    public void setExpirationDate(LocalDateTime expirationDate) {
        this.expirationDate = expirationDate;
    }

    public boolean isOrganic() {
        return isOrganic;
    }

    public void setOrganic(boolean organic) {
        isOrganic = organic;
    }

    // Override the displayInfo method to include additional information
    @Override
    public void displayInfo() {
        super.displayInfo(); // Call the superclass method
        System.out.println("Expiration Date: " + expirationDate + ", Organic: " + isOrganic);
    }
}
